Source/Parallelization/WarpXComm_K.H

```diff
@@ -44,18 +44,18 @@ void warpx_interp (int j, int k, int l,
 
     // Refinement ratio
     const int rj = rr[0];
-    const int rk = (AMREX_SPACEDIM == 1) ? 1 : rr[1];
-    const int rl = (AMREX_SPACEDIM <= 2) ? 1 : rr[2];
+    const int rk = (AMREX_SPACEDIM > 1) ? rr[1] : 1;
+    const int rl = (AMREX_SPACEDIM > 2) ? rr[2] : 1;
 
     // Staggering (0: cell-centered; 1: nodal)
     const int sj = arr_stag[0];
-    const int sk = (AMREX_SPACEDIM == 1) ? 0 : arr_stag[1];
-    const int sl = (AMREX_SPACEDIM <= 2) ? 0 : arr_stag[2];
+    const int sk = (AMREX_SPACEDIM > 1) ? arr_stag[1] : 1;
+    const int sl = (AMREX_SPACEDIM > 2) ? arr_stag[2] : 1;
 
     // Number of points used for interpolation from coarse grid to fine grid
     const int nj = 2;
-    const int nk = 2;
-    const int nl = 2;
+    const int nk = (AMREX_SPACEDIM > 1) ? 2 : 1;
+    const int nl = (AMREX_SPACEDIM > 2) ? 2 : 1;
```

Comment:
This changes the behavior of the code. Is this intended?

Reply:
Yes, this is also intended. In the original version is was setting the `nk = 2` which meant that it would have a loop of length 2 over the unused direction. Following the math, the value added to `res` was zero for the second iteration of the loop so it was wasted computation.